ambry-api/src/main/java/com.github.ambry/router/ReadableStreamChannel.java

```diff
@@ -45,4 +46,33 @@
    * @return the {@link Future} that will eventually contain the result of the operation.
    */
   public Future<Long> readInto(AsyncWritableChannel asyncWritableChannel, Callback<Long> callback);
+
+  /**
+   * Set the digest algorithm to use on the data that is being streamed from the channel. Once the channel is emptied,
+   * the digest can be obtained via {@link #getDigest()}.
+   * <p/>
+   * This function is ideally called before {@link #readInto(AsyncWritableChannel, Callback)}. After a call to
```

Comment:
As discussed, we will keep this for now with the implementation docs providing more detail.